tipgstac/models.py

```diff
@@ -78,6 +78,63 @@ def validate_bbox(cls, v: BBox):
         return v
 
 
+class CollectionsSearch(BaseModel):
+    """PgSTAC Collections Search Query model."""
+
+    ids: Optional[List[str]] = None
+    bbox: Optional[BBox] = None
+    intersects: Optional[Geometry] = None
+    query: Optional[Dict[str, Dict[Operator, Any]]] = None
+    filter: Optional[Dict] = None
+    datetime: Optional[str] = None
+    sortby: Optional[Any] = None
+    fields: Optional[Dict[str, Set]] = None
+    filter_lang: Annotated[Optional[FilterLang], Field(alias="filter-lang")] = None
+    limit: Optional[int] = None
+    offset: Optional[int] = None
+    conf: Optional[Dict] = None
+
+    model_config = {"extra": "ignore"}
+
+    @field_validator("intersects")
+    def validate_spatial(cls, v: Optional[Geometry], info: ValidationInfo):
+        """Make sure bbox is not used with Intersects."""
+        if v and info.data["bbox"]:
+            raise ValueError("intersects and bbox parameters are mutually exclusive")
+
+        return v
+
+    @field_validator("bbox")
+    def validate_bbox(cls, v: BBox):
+        """Validate BBOX."""
+        if v:
+            # Validate order
+            if len(v) == 4:
+                xmin, ymin, xmax, ymax = v
+            else:
+                xmin, ymin, min_elev, xmax, ymax, max_elev = v
+                if max_elev < min_elev:
+                    raise ValueError(
+                        "Maximum elevation must greater than minimum elevation"
+                    )
+
+            if xmax < xmin:
+                raise ValueError(
+                    "Maximum longitude must be greater than minimum longitude"
+                )
+
+            if ymax < ymin:
+                raise ValueError(
+                    "Maximum longitude must be greater than minimum longitude"
+                )
+
+            # Validate against WGS84
+            if xmin < -180 or ymin < -90 or xmax > 180 or ymax > 90:
+                raise ValueError("Bounding box must be within (-180, -90, 180, 90)")
+
+        return v
```

Comment:
we should externalize those validators because they are used in the ItemSearch model too